src/app/wallets/get-on-chain-fee.ts

```diff
@@ -0,0 +1,103 @@
+import { TxDecoder } from "@domain/bitcoin/onchain"
+import { WalletsRepository } from "@services/mongoose"
+import { OnChainService } from "@services/lnd/onchain-service"
+import { BTC_NETWORK, getOnChainWalletConfig } from "@config/app"
+import { checkedToSats, checkedToTargetConfs } from "@domain/bitcoin"
+import {
+  CouldNotFindError,
+  LessThanDustThresholdError,
+  InsufficientBalanceError,
+} from "@domain/errors"
+import { getBalanceForWalletId } from "./get-balance-for-wallet"
+
+const { dustThreshold } = getOnChainWalletConfig()
+
+export const getOnChainFee = async ({
+  wallet,
+  amount,
+  address,
+  targetConfirmations,
+}: GetOnChainFeeArgs): Promise<Satoshis | ApplicationError> => {
+  const walletsRepo = WalletsRepository()
+  const payeeWallet = await walletsRepo.findByAddress(address)
+  if (payeeWallet instanceof CouldNotFindError) {
+    if (amount < dustThreshold) {
+      return new LessThanDustThresholdError(
+        `Use lightning to send amounts less than ${dustThreshold}`,
+      )
+    }
+
+    const balance = await getBalanceForWalletId(wallet.id)
+    if (balance instanceof Error) return balance
+
+    // avoids lnd balance sniffing attack
+    if (balance < amount) {
+      return new InsufficientBalanceError("Balance is too low")
+    }
+
+    const onChainService = OnChainService(TxDecoder(BTC_NETWORK))
+    if (onChainService instanceof Error) return onChainService
+
+    const onChainFee = await onChainService.getOnChainFeeEstimate(
+      amount,
+      address,
+      targetConfirmations,
+    )
+    if (onChainFee instanceof Error) return onChainFee
+
+    return (onChainFee + wallet.withdrawFee) as Satoshis
+  }
+
+  if (payeeWallet instanceof Error) return payeeWallet
+
+  // intraledger tx has zero fee
+  return 0 as Satoshis
```

Comment:
This is business logic and should not be hard coded into the use-case method. Even if its trivial its best to be explicit like with the [DepositFeeCaluclator](https://github.com/GaloyMoney/galoy/blob/main/src/domain/wallets/deposit-fee-calculator.ts)

@vindard did you have some kind of business logic code that handles fee in your `pay` method refactoring?